model/src/main/java/org/openremote/model/webhook/Webhook.java

```diff
@@ -0,0 +1,111 @@
+package org.openremote.model.webhook;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class Webhook {
+
+    public enum Method {
+        GET, POST, PUT, DELETE
+    }
+
+    public enum AuthMethod {
+        NONE, HTTP_BASIC, API_KEY, OAUTH2
+    }
+
+    public static class AuthDetails {
+        public String username;
+        public String password;
+        public String apiKey;
+        public Method method;
+        public String url;
+        public String clientId;
+        public String clientSecret;
+    }
+
+    public static class Header {
+        public String header;
+        public String value;
+    }
+
+    /* --------- */
+
+    protected String name;
+    protected String url;
+    protected Header[] headers;
+    protected Method method;
+    protected AuthMethod authMethod;
+    protected AuthDetails authDetails;
```

Comment:
As mentioned above these can be replaced with `OAuthGrant` and `UsernamePassword` see `Agent` class